### DISCUSSION TITLE
Feature request: Add support for go lang linters and lsp

### DISCUSSION THREAD

**Discussion by legend-sabbir** [mkslanc/ace-linters#62] [General]
Is it possible to add support for `Go lang`  in browser mode ?

**Comment by mkslanc** [comment]
If there is any WASM golang linter or implemented golang linter in javascript it's possible

**Reply by legend-sabbir** [reply]
then i don't think its possible . Anyway can you guide me on how to implement gopls using a server in ace editor ? Android devices has a terminal `termux` it is good . There is some limitations but still it is very good .

**Reply by legend-sabbir** [reply]
Also one issue
on v 0.7.1 it was totally fine


https://github.com/mkslanc/ace-linters/assets/105509363/7dde92eb-af4b-4da5-b389-1cfe6d4b3b90

on 0.10.3 or maybe after all the versions of 0.7.1


https://github.com/mkslanc/ace-linters/assets/105509363/d016fa40-bf8a-4a3a-912f-bf5d401b65ef

It is saying i redeclared the variables . Does it save sessions ? the 0.7.1 was good

**Reply by mkslanc** [reply]
I made attempt to wrap gopls with WebSocket, but there seems need a lot more effort to figure out, why after handshake it's sending empty responses, but if you want to investigate - https://github.com/mkslanc/goplsws/blob/master/ws.js (this is draft server). You could take any websockets example and connect by 3030 port

**Reply by mkslanc** [reply]
I think I solved second issue in new release, you will need to call new API endpoint [closeDocument](https://github.com/mkslanc/ace-linters/blob/main/packages/ace-linters/language-provider.ts#LL243C7-L243C7) to remove current session with current content from all linked services

**Reply by legend-sabbir** [reply]
`closeDocument` is problematic when switching between sessions otherwise works fine .
Example of switching file 

https://github.com/mkslanc/ace-linters/assets/105509363/792f40e1-5b3a-4f5f-8cf2-e1e8206039ea

```javascript
editor.on("changeSession", ({oldSession}) => {
    provider.closeDocument(oldSession)
})
```

**Reply by mkslanc** [reply]
In that case you shouldn't close on change session, you should close on implicit `onclose` tab event (if there such thing in acode)

**Reply by legend-sabbir** [reply]
I can do that , but while two tabs are open and user is switching between sessions it remembers prevoius sessions data and  It gives the same error as before , if you want i can record video, 

offtopic: i created bracket pair colorizer for some modes , it works but im pretty sure thats not how it should be implemented , if you want i can share code for that

**Reply by mkslanc** [reply]
Hmm, if it's same editor and session changed it should register new session every time, could you try implicitely do -
```javascript 
editor.on("changeSession", ({oldSession, session}) => {
            provider.closeDocument(oldSession);
            provider.$registerSession(session)
        });
```
? 
For bracket pair colorizer would be better to create PR in [main ace repo ](https://github.com/ajaxorg/ace/pulls)

**Reply by legend-sabbir** [reply]
```javascript 
editor.on("changeSession", ({oldSession, session}) => {
            provider.closeDocument(oldSession);
            provider.$registerSession(session)
        }); 
```
 it created even more problems .

**Comment by legend-sabbir** [comment]
This issue is fixed, I dont know how
